### PR TITLE
update flash algo for imxrt1021xxxxx

### PR DESCRIPTION
**Pull request by Hoohaha**
Update flash algo to fix flash erasing problem if evk-mimxrt1021 in serial download mode.

### CODE REVIEW

**Comment by Hoohaha** [comment]
@flit can you help review this?

**Comment by flit** [comment]
@Hoohaha Really sorry once again! Remember you can ping me via email in case I missed a notification. I'm also going to try to change how I'm watching notifications so I hopefully don't miss so many.

File: pyocd/target/family/target_imxrt.py [hunk 1]
```diff
@@ -82,8 +84,19 @@ def set_reset_catch(self, reset_type=None):
...
         bootdevice = (self.read_memory(CortexM7_IMXRT.SRC_SBMR1) & 0x000000F0) >> 4
-        LOG.info("IMXRT Boot mode: %s" % self.BOOT_MODES[bootmode])
+        LOG.info("IMXRT Boot Device: %x" % bootdevice)
+        return bootdevice
+
+    def _get_flash_vector_addr(self):
+        mem = self.memory_map.get_boot_memory()
+        if mem and mem.type == MemoryType.FLASH:
+            return mem.start + 0x1004
+        return None
```
Comment (flit) [inline pyocd/target/family/target_imxrt.py:94]: The `return None` is unnecessary since None is the default return value from all functions.

**Review by flit — APPROVED** [review]

**Pull request merged**